pkg/controller/infrastructure/actuator.go

```diff
@@ -294,6 +292,40 @@ func (a *actuator) isWhitelistedImageID(imageID string) bool {
 	return false
 }
 
+// ensureOldSSHKeyDetached ensures the compatibility when ssh key is generated in the current cluster via terraform.
+func (a *actuator) ensureOldSSHKeyDetached(ctx context.Context, infra *extensionsv1alpha1.Infrastructure) error {
+
+	if infra.Status.ProviderStatus == nil {
+		return nil
+	}
+
+	infrastructureStatus := &alicloudv1alpha1.InfrastructureStatus{}
+	if _, _, err := a.Decoder().Decode(infra.Status.ProviderStatus.Raw, nil, infrastructureStatus); err != nil {
+		return err
+	}
+
+	// nolint
```

Comment:
What is the reason this `// no lint` comment? `make check` is successful even without it.

Reply:
I originally annotate the KeyPairName to be `deprecated`. But failed in generated file. I will remove this.